plugins/CorePluginsAdmin/API.php

```diff
@@ -61,15 +64,46 @@ public function setSystemSettings($settingValues, $passwordConfirmation = false)
 
         $this->settingsMetadata->setPluginSettings($pluginsSettings, $settingValues);
 
+        $sendSettingsChangedNotificationEmailPlugins = [];
+
         try {
             foreach ($pluginsSettings as $pluginSetting) {
                 if (!empty($settingValues[$pluginSetting->getPluginName()])) {
                     $pluginSetting->save();
+
+                    $pluginName = $pluginSetting->getPluginName();
+                    if (in_array($pluginName, array_keys(SecurityNotificationEmail::$notifyPluginList))) {
+                        $sendSettingsChangedNotificationEmailPlugins[] = $pluginName;
+                    }
                 }
             }
         } catch (Exception $e) {
             throw new Exception(Piwik::translate('CoreAdminHome_PluginSettingsSaveFailed'));
         }
+
+        foreach ($sendSettingsChangedNotificationEmailPlugins as $plugin) {
```

Comment:
in this loop, it's possible to send way too many emails, if there are more superusers in the system. do we still want to send on-the-fly? or use a job here?

Reply:
We could reduce the number of emails by sending all the plugin names in a single email. (Ie, "There were changes made to the following system settings: X, Y, and Z.")

There could still be an edge case of too many superusers, I suppose. We could just send one email to multiple recipeints. Ie, using `Mail::addTo()` for each superuser. If that results in just one mail() call or SMTP request, I think it might be ok.

Setting up a queue and a task to process it for mails, however, seems like too much effort for this ticket.

(cc @tsteur)

Reply:
I reduced the number of emails by sending only one that contains all the setting groups that changed and sending one email to superusers using the `addTo()`.